lib/brakeman/checks/check_forgery_setting.rb

```diff
@@ -7,67 +7,69 @@
 class Brakeman::CheckForgerySetting < Brakeman::BaseCheck
   Brakeman::Checks.add self
 
-  @description = "Verifies that protect_from_forgery is enabled in ApplicationController"
+  @description = "Verifies that protect_from_forgery is enabled in direct subclasses of ActionController::Base"
 
   def run_check
-    app_controller = tracker.controllers[:ApplicationController]
-    return unless app_controller and app_controller.ancestor? :"ActionController::Base"
+    tracker.controllers
+        .select { |_, controller| controller.parent == :"ActionController::Base" }
+        .each do |name, controller|
 
-    if tracker.config.allow_forgery_protection?
-      warn :controller => :ApplicationController,
-        :warning_type => "Cross-Site Request Forgery",
-        :warning_code => :csrf_protection_disabled,
-        :message => "Forgery protection is disabled",
-        :confidence => CONFIDENCE[:high],
-        :file => app_controller.file
+      if tracker.config.allow_forgery_protection?
+        warn :controller => name,
```

Comment:
I factored out that check from the `.each` iteration. One side effect of that is that for the `rails3.rb` test, we prioritize the warning of CVE20110447 over the missing `protect_from_forgery` call. See https://github.com/presidentbeef/brakeman/pull/858/files#diff-29d5ade8f0be60f47a3073ebab75b8d4L386. Which kind of makes sense in my opinion because there's no point asking the user to add the call to when rails need upgrading or patching in the first place.